### PR TITLE
Wrap Linux browser launches with dbus-launch when possible

### PR DESCRIPTION
**Pull request by davidhao3300**
I noticed that Chrome sometimes fails to start properly on Linux VMs. Recently, a solution has been found here: https://github.com/angular/protractor/issues/2419 that solves the issue (I've confirmed that by running tests repeatedly and getting no Chrome hangs). I implemented their solution here. This first checks whether the requisite executable is installed before wrapping the browsers, so this shouldn't cause any regressions.

### CODE REVIEW

**Comment by davidhao3300** [comment]
not sure how to test this

**Comment by johanneswuerbach** [comment]
Thanks :+1:, looks like your change broke CI mind taking a look?

**Comment by davidhao3300** [comment]
yeah, I'll get around to it today.

**Comment by davidhao3300** [comment]
Okay, all checks have passed

File: lib/browser_launcher.js [hunk 1]
```diff
@@ -239,6 +240,19 @@ function browsersForPlatform() {
         supported: findableByWhichOrModule
       }
     ];
+
+    // If dbus-launch is present, wrap google-chrome launch
+    // to prevent a race condition in Google Chrome
+    // Discussion here: https://github.com/angular/protractor/issues/2419
+    if (whereis('dbus-launch')) {
```
Comment (johanneswuerbach) [inline lib/browser_launcher.js]: We already have a function, which does this. Could you replace `node-whereis` with https://github.com/testem/testem/blob/master/lib/fileutils.js#L19?

Comment (davidhao3300) [inline lib/browser_launcher.js]: sure

**Comment by davidhao3300** [comment]
Alright, changes have been made. I noticed there was a case where dbus-launch could be installed but not the original browser. This now handles that by first checking the availability of the original browser before wrapping the browser in dbus-launch.

**Comment by johanneswuerbach** [comment]
Thanks :+1:

**Pull request merged**